packages/dds/tree/src/test/feature-libraries/defaultFieldKinds.spec.ts

```diff
@@ -0,0 +1,281 @@
+        assert.deepEqual(decoded, change1WithChildChange);
+    });
+});
+
+describe("Optional field changesets", () => {
+    const fieldHandler: FieldChangeHandler<FieldKinds.OptionalChangeset> = FieldKinds.optional.changeHandler;
+    const editor: FieldKinds.OptionalFieldEditor = fieldHandler.editor as FieldKinds.OptionalFieldEditor;
+
+    const change1: FieldKinds.OptionalChangeset = {
+        fieldChange: { newContent: tree1, wasEmpty: true },
+        childChange: nodeChange1,
+    };
+
+    const change2: FieldKinds.OptionalChangeset = editor.set(singleTextCursor(tree2), false);
+    const change3: FieldKinds.OptionalChangeset = editor.set(singleTextCursor(tree2), true);
+
+    it("can be composed", () => {
+        const childComposer = (_: NodeChangeset[]) => assert.fail("Should not be called");
+        const composed = fieldHandler.rebaser.compose([change1, change2], childComposer);
+        assert.deepEqual(composed, change3);
+    });
+
+    it("can compose child changes", () => {
+        const childChange: FieldKinds.OptionalChangeset = editor.buildChildChange(0, nodeChange2);
+        const expected: FieldKinds.OptionalChangeset = {
+            fieldChange: change1.fieldChange,
+            childChange: nodeChange3,
+        };
+
+        assert.deepEqual(
+            fieldHandler.rebaser.compose([change1, childChange], childComposer1_2),
+            expected,
+        )
+    });
+
+    it("can be inverted", () => {
+        const childInverter = (change: NodeChangeset) => {
+            assert.deepEqual(change, nodeChange1);
+            return nodeChange2;
+        };
+
+        const expected: FieldKinds.OptionalChangeset = {
+            fieldChange: { wasEmpty: false },
+            childChange: nodeChange2,
+        };
+
+        assert.deepEqual(
+            fieldHandler.rebaser.invert(change1, childInverter),
+            expected,
+        );
+    });
+
+    it("can be rebased", () => {
+        const childRebaser = (_change: NodeChangeset, _base: NodeChangeset) => assert.fail("Should not be called");
+        assert.deepEqual(
+            fieldHandler.rebaser.rebase(change3, change1, childRebaser),
+            change2,
+        );
+    });
+
+    it("can rebase child change", () => {
+        const baseChange: FieldKinds.OptionalChangeset = { childChange: nodeChange1 };
+        const changeToRebase: FieldKinds.OptionalChangeset = { childChange: nodeChange2 };
+
+        const childRebaser = (change: NodeChangeset, base: NodeChangeset) => {
+            assert.deepEqual(change, nodeChange2);
+            assert.deepEqual(base, nodeChange1);
+            return nodeChange3;
+        };
+
+        const expected: FieldKinds.OptionalChangeset = { childChange: nodeChange3 };
+
+        assert.deepEqual(
+            fieldHandler.rebaser.rebase(changeToRebase, baseChange, childRebaser),
+            expected,
+        );
+    });
+
+    it("can be converted to a delta when field was empty", () => {
```

Comment:
It would be nice to have a test case for the scenario where the field is neither being set nor cleared but there are changes below it.